compiler/hash-parser/src/gen.rs

```diff
@@ -173,6 +173,18 @@ where
         }
     }
 
+    /// Get the next location of the token, if there is no token after, we use the 
```

Comment:
Maybe "get the location of the next token", rather than "get the next location of the token"?